### PR TITLE
Update dart2js.md

### LINKED ISSUE #1824: dart2js: recommend use of `-O` flag (as of Dart 2.1)
From https://github.com/dart-lang/site-www/issues/1164#issuecomment-423501121:

>> #### dart2js
>> * Added `-O` flag to tune optimization levels.  ...
>>
>>    We recommend to enable optimizations using the `-O` flag **instead of individual flags** for each optimization. This is because the `-O` flag is intended to be stable and continue to work in future versions of dart2js, ... At this time we recommend to test and debug with `-O1` and to deploy with `-O3`.
>
> (Emphasis mine) We mention use of the individual optimization flags on webdev, so any such prose would need to be updated. I'll create a separate issue to track that.

### PR DESCRIPTION
**Pull request by kwalrath**
Fixes #1824.

Staged: https://kw-webdev-dartlang-1.firebaseapp.com/tools/dart2js

I switched to `-o` from `--output=` because there's a bug in the styles that makes the `=` invisible.

Note: `--package-root` is mentioned by the help for `--packages`, but isn't itself in the dart2js help.

### CODE REVIEW

File: src/tools/dart2js.md [hunk 1]
```diff
@@ -24,13 +24,21 @@ on debugging the JavaScript that dart2js generates.
...
 ```
 
 This command produces a file that contains the JavaScript equivalent of your
 Dart code. It also produces a source map, which can help you debug the
 JavaScript version of the app more easily.
 
+<aside class="alert alert-info" markdown="1">
+  **Note:**
+  The <code>-O<em>n</em></code> argument specifies the optimization level.
+  We recommend using `-O1` (the default) during development and
```
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/tools/dart2js.md]: Hmmm... Should we say that we recommend using dartdevc for rapid development, with dart2js -O1 for tests? Then we recommend deploying with -O3, as long as their testing shows that their app doesn't crash?

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/tools/dart2js.md]: We already cover dartdevc above, so how about:

  We recommend starting at `-O1` (the default), and increasing to
  `-O3` when you're ready to deploy.

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/tools/dart2js.md]: Now I'm thinking:

We recommend starting at -O1 (the default) and then increasing to -O2 or higher when you’re ready to deploy. The -O3 and -O4 optimization levels are suitable only for well tested code (see the -On descriptions, below).

**Comment by kwalrath** [comment]
ptal. I've updated the staged version: https://kw-webdev-dartlang-1.firebaseapp.com/tools/dart2js

**Comment by kwalrath** [comment]
I'm going to merge this, but please let me know if you'd like any tweaks.

File: src/tools/dart2js.md [hunk 2]
```diff
@@ -42,182 +50,109 @@ For more information, see the [build_web_compilers README.][build_web_compilers]
...
 
 Some other handy options include:
 
 `--packages=<path>`
-: Specify the path to the package resolution configuration file.
+: Specifies the path to the package resolution configuration file.
   For more information, see
-  [Package Resolution Configuration File](https://github.com/lrhn/dep-pkgspec/blob/master/DEP-pkgspec.md).
+  [Package Resolution Configuration File.](https://github.com/lrhn/dep-pkgspec/blob/master/DEP-pkgspec.md)
   _This option cannot be used with `--package-root`._
```
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/tools/dart2js.md]: We should probably remove the mention from the `dart2js -vh` text, as well. I've filed dart-lang/sdk#35393.

File: src/tools/dart2js.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sigmundch) [reply src/tools/dart2js.md]: nice

File: src/tools/dart2js.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sigmundch) [reply src/tools/dart2js.md]: thanks!

**Comment by sigmundch** [comment]
Fantastic! LGTM

**Pull request merged**